src/nodes/parameter.h

```diff
@@ -352,11 +352,47 @@ template <typename T> class PointerParameter : public Parameter<T>
 
     public:
     // Set the object
-    void set(const T &value) override{};
+    void set(const T &value) override {};
     // Assign the value of another parameter to this one.
     bool assign(ParameterBase *other) override { return false; }
 };
 
+// OptionalPointerParameter, returning a pointer from a target object rather than the object itself
+template <typename T> class OptionalPointerParameter : public Parameter<T>
+{
+    public:
+    OptionalPointerParameter(Node *parent, std::string_view name, std::string_view description, std::optional<T *> &object)
+        : Parameter<T>(parent, name, description, object_)
+    {
+        object_ = object;
+    }
+    virtual ~OptionalPointerParameter() = default;
+
+    /*
+     * Data
+     */
+    protected:
+    // Optional object
+    std::optional<T *> &object_;
+
+    public:
+    // Return the parameter value
+    T &get()
+    {
+        if (object_.has_value())
+            return object_.value();
+
+        return {};
```

Comment:
I think here we can do:
```suggestion
        return object_.value_or(nullptr);
```